optuna/trial/_trial.py

```diff
@@ -127,7 +127,12 @@ def objective(trial):
             low:
                 Lower endpoint of the range of suggested values. ``low`` is included in the range.
             high:
-                Upper endpoint of the range of suggested values. ``high`` is included in the range.
+                Upper endpoint of the range of suggested values. ``high`` is excluded from the
+                range.
+
+                .. note::
+                    If ``step`` is specified, ``high`` is included as well as ``low``.
+
```

Comment:
I don't think this change is related to the PR title. Could you revert it, please?